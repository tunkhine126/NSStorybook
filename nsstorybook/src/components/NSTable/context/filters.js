import React, { useState, createContext } from 'react';
import PropTypes from 'prop-types';

export const NSFiltersContext = createContext();

export const NSFiltersProvider = ({
  children,
  filterCategories,
  lastUpdateProps,
  dateRangeProps,
  handleSelected,
}) => {
  const types = {
    multi: {},
    levels: {},
    single: {},
    date: {},
  };

  const { lastUpdate } = lastUpdateProps;

  ['multi', 'levels', 'single', 'date'].forEach(type => {
    filterCategories
      .filter(c => c.type === type)
      .forEach(({ id }) => {
        types[type][id] = [];
      });
  });

  const [singleSelectOptions, setSingleSelectOptions] = useState(types.single);
  const [multiSelectOptions, setMultiSelectOptions] = useState(types.multi);
  const [levelSelectOptions, setLevelSelectOptions] = useState(types.levels);
  const [showCalendar, setShowCalendar] = useState(
    lastUpdate.query === 'custom'
  );
  const [localDateRange, setlocalDateRange] = useState({
    before: null,
    after: null,
  });

  return (
    <NSFiltersContext.Provider
      value={{
        multiSelect: { multiSelectOptions, setMultiSelectOptions },
        levelSelect: { levelSelectOptions, setLevelSelectOptions },
        singleSelect: { singleSelectOptions, setSingleSelectOptions },
        calendarProps: { showCalendar, setShowCalendar },
        localDateProps: { localDateRange, setlocalDateRange },
        lastUpdateProps,
        dateRangeProps,
        filterCategories,
        handleSelected,
      }}
    >
      {children}
    </NSFiltersContext.Provider>
  );
};

NSFiltersProvider.propTypes = {
  children: PropTypes.any.isRequired,
  dateRangeProps: PropTypes.object.isRequired,
  lastUpdateProps: PropTypes.object.isRequired,
  filterCategories: PropTypes.array.isRequired,
  handleSelected: PropTypes.func.isRequired,
};
