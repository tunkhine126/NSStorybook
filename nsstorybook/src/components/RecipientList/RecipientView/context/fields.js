import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

export const FieldsContext = createContext();

export const FieldsProvider = ({ children, ...rest }) => {
  const [activePhotoField, setActivePhotoField] = useState(null);
  const [recipientPhoto, setRecipientPhoto] = useState({});

  return (
    <FieldsContext.Provider
      value={{
        activePhotoProps: { activePhotoField, setActivePhotoField },
        recipientPhotoProps: { recipientPhoto, setRecipientPhoto },
        rest,
      }}
    >
      {children}
    </FieldsContext.Provider>
  );
};

FieldsProvider.propTypes = {
  children: PropTypes.any.isRequired,
};
