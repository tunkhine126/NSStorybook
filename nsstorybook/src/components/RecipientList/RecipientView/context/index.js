import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

export const RecipientViewContext = createContext();

export const RecipientViewProvider = ({
  children,
  tabChange,
  rdData,
  data,
  profileDetails,
  primaryRecipient,
  match,
  ...rest
}) => {
  const [toaster, setToaster] = useState({
    show: false,
    recipientUuid: null,
    field: null,
    undoHandler: null,
  });
  const [saveState, setSaveState] = useState({});

  return (
    <RecipientViewContext.Provider
      value={{
        data,
        rdData,
        tabChange,
        profileDetails,
        primaryRecipient,
        match,
        toasterProps: { toaster, setToaster },
        saveProps: { saveState, setSaveState },
        rest,
      }}
    >
      {children}
    </RecipientViewContext.Provider>
  );
};

RecipientViewProvider.propTypes = {
  children: PropTypes.any.isRequired,
  tabChange: PropTypes.func.isRequired,
  profileDetails: PropTypes.object.isRequired,
  primaryRecipient: PropTypes.bool.isRequired,
  data: PropTypes.object.isRequired,
  rdData: PropTypes.any.isRequired,
  match: PropTypes.any.isRequired,
};
