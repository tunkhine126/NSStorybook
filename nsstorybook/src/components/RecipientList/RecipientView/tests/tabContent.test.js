/**
 *
 * Tests for TabContent
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import TestingComponent from 'components/shared/Testing';
import { render } from '@testing-library/react';
// import 'jest-dom/extend-expect'; // add some helpful assertions
import TabContent from '../tabContent';
import { RecipientViewProvider } from '../context';

describe('<TabContent />', () => {
  const props = {
    profileDetails: {
      recipientName: 'Country',
      childRecipients: 'Communities',
    },
    data: {
      name: 'Test',
      avatarUrl: null,
      internalId: '3erf23-r323d-gewf3-233',
      children: [],
      recipientDefinition: {
        customFields: [],
      },
    },
    match: {
      isExact: true,
      params: { id: '07c538ec-f5c8-11e9-a1ad-22000be1c01d' },
      path: '/families/family-list/family/:id',
      url: '/families/family-list/family/07c538ec-f5c8-11e9-a1ad-22000be1c01d',
    },
    rdData: {},
    tabChange: () => {},
  };

  const details = {
    addRecipientName: 'New Country',
    columns: [
      {
        id: 'recipientName',
        defaultWidth: '380px',
        label: 'Family',
        align: 'left',
        type: 'single',
        columnData: null,
        noData: false,
        disablePadding: true,
        show: true,
        sortable: true,
        defaultFilter: false,
        filter: false,
        custom: false,
      },
      {
        id: 'id',
        defaultWidth: '450px',
        label: 'Family ID',
        align: 'left',
        type: 'single',
        columnData: null,
        noData: false,
        disablePadding: false,
        show: true,
        sortable: true,
        filter: false,
        defaultFilter: false,
        custom: false,
      },
      {
        id: 'completedSurveys',
        defaultWidth: '123px',
        type: 'multi',
        label: 'Completed Surveys',
        filterWidth: 170,
        align: 'center',
        columnData: null,
        disablePadding: true,
        noData: false,
        show: true,
        sortable: false,
        filter: true,
        defaultFilter: true,
        custom: false,
      },
      {
        id: 'childRecipients',
        defaultWidth: '100px',
        type: 'multi',
        label: 'Family Members',
        align: 'center',
        columnData: null,
        noData: false,
        disablePadding: true,
        show: true,
        sortable: false,
        filter: false,
        defaultFilter: false,
        custom: false,
      },
      {
        id: 'parentRecipients',
        type: 'levels',
        label: 'Community',
        filterWidth: 150,
        align: 'center',
        noData: false,
        columnData: null,
        disablePadding: true,
        sortable: false,
        show: true,
        filter: true,
        noColumn: true,
        defaultFilter: true,
        custom: false,
      },
      {
        id: 'lastUpdate',
        defaultWidth: '180px',
        type: 'date',
        label: 'Last Update',
        filterWidth: 150,
        align: 'center',
        columnData: null,
        noData: true,
        disablePadding: true,
        show: true,
        sortable: true,
        filter: true,
        defaultFilter: true,
        custom: false,
      },
    ],
    title: 'Countries',
    recipientDefintionUuid: '60577bb8-5641-11e9-9966-22000bd4493b',
    childRoutes: [],
    hierarchyId: 3,
    hasParentRecipient: false,
    nestedRecipientTree: [],
  };

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');

    render(
      <TestingComponent>
        <RecipientViewProvider {...props} primaryRecipient>
          <TabContent value={0} details={details} indexChange={() => {}} />
        </RecipientViewProvider>
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestingComponent>
        <RecipientViewProvider {...props} primaryRecipient={false}>
          <TabContent value={0} details={details} indexChange={() => {}} />
        </RecipientViewProvider>
      </TestingComponent>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
