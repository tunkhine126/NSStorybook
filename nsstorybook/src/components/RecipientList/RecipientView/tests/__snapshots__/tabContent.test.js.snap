// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabContent /> Should render and match the snapshot 1`] = `
<div
  class="styles__TabContentStyle-qa8k4k-8 kybHyw"
>
  <div
    style="overflow-x: hidden;"
  >
    <div
      class="react-swipeable-view-container"
      style="flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform;"
    >
      <div
        aria-hidden="false"
        data-swipeable="true"
        style="width: 100%; flex-shrink: 0; overflow: auto;"
      >
        <div
          class="MuiGrid-root makeStyles-container-279 MuiGrid-container MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-space-between"
        >
          <div
            class="MuiGrid-root makeStyles-column1-289 MuiGrid-item"
          >
            <div
              class="MuiPaper-root makeStyles-paper-317 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <p
                class="MuiTypography-root makeStyles-widgetHeader-328 makeStyles-widgetPadding-329 MuiTypography-body1 MuiTypography-gutterBottom"
              >
                Communities
              </p>
              <hr
                class="MuiDivider-root MuiDivider-light"
              />
              <div
                class="styles__ExpansionPanelStyle-qa8k4k-9 iyqZhl"
              >
                <hr
                  class="MuiDivider-root MuiDivider-light"
                />
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-center"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-addBtn-277 MuiButton-textPrimary makeStyles-textPrimary-502"
                    style="margin-right: auto; pointer-events: none; opacity: 0.5;"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label makeStyles-label-501"
                    >
                      <span>
                        Add Recipient
                      </span>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root makeStyles-column2-290 MuiGrid-item"
          >
            <div
              class="MuiPaper-root makeStyles-paper-317 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <p
                class="MuiTypography-root makeStyles-widgetHeader-328 makeStyles-widgetPadding-329 MuiTypography-body1 MuiTypography-gutterBottom"
              >
                Snapshot
              </p>
              <hr
                class="MuiDivider-root MuiDivider-light"
              />
              <div
                class="styles__ExpansionPanelStyle-qa8k4k-9 iyqZhl"
              >
                <hr
                  class="MuiDivider-root MuiDivider-light"
                />
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-center"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-addBtn-277 MuiButton-textPrimary makeStyles-textPrimary-536"
                    style="margin-right: auto; pointer-events: initial;"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label makeStyles-label-535"
                    >
                      <span>
                        Full Profile
                      </span>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root makeStyles-activityContainer-276 MuiGrid-item"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
